backend/apps/github/models/repository.py

```diff
@@ -234,74 +235,80 @@ def from_github(
-        # License.
+        # Process optional data sources
+        optional_data = [
+            (commits, self._process_commits_data),
+            (contributors, self._process_contributors_data),
+            (languages, self._process_languages_data),
+        ]
+        for data, processor in optional_data:
+            if data is not None:
+                processor(data)
+
+        # License
         self.license = gh_repository.license.spdx_id if gh_repository.license else ""
 
-        # Fetch project metadata from funding.yml file.
+        # Funding metadata
+        self._process_funding_data(gh_repository)
+
+        # Foreign keys
+        self.organization = organization
+        self.owner = user
+
+    def _process_commits_data(self, commits):
+        """Process commits data."""
+        try:
+            self.commits_count = commits.totalCount
+        except GithubException as e:
+            if e.data["status"] == "409" and "Git Repository is empty" in e.data["message"]:
+                self.is_empty = True
+
+    def _process_contributors_data(self, contributors):
+        """Process contributors data."""
+        self.contributors_count = contributors.totalCount
+
+    def _process_languages_data(self, languages):
+        """Process languages data."""
+        total_size = sum(languages.values())
+        self.languages = {
+            language: round(size * 100.0 / total_size, 1) for language, size in languages.items()
+        }
+
+    def _process_funding_data(self, gh_repository):
+        """Process funding.yml data."""
         try:
             funding_yml = gh_repository.get_contents(".github/FUNDING.yml")
             yaml_content = b64decode(funding_yml.content).decode()
             self.funding_yml = yaml.safe_load(yaml_content)
             self.has_funding_yml = True
 
-            # Set funding policy compliance flag.
-            is_funding_policy_compliant = True
-            for platform, targets in self.funding_yml.items():
-                for target in targets if isinstance(targets, list) else [targets]:
-                    if not target:
-                        continue
-                    is_funding_policy_compliant = check_funding_policy_compliance(
-                        platform,
-                        target,
-                    )
-
-                    if not is_funding_policy_compliant:
-                        break
-
-                if not is_funding_policy_compliant:
-                    break
-            self.is_funding_policy_compliant = is_funding_policy_compliant
+            # Check funding policy compliance
+            self.is_funding_policy_compliant = self._check_all_funding_compliance()
         except (AttributeError, GithubException):
             self.has_funding_yml = False
             self.is_funding_policy_compliant = True
 
-        # FKs.
-        self.organization = organization
-        self.owner = user
+    def _check_all_funding_compliance(self) -> bool:
```

Comment:
Maybe it'd be better to refactor `check_funding_policy_compliance` function to accept an iterable instead of a single target?